tests/test_utils.py

```diff
@@ -409,14 +410,19 @@ def test_get_dict_from_mapping(self):
             "match_string": "",
             "subject_preprocessing": "",
             "object_preprocessing": "",
-            "semantic_similarity_score": np.nan,
-            "semantic_similarity_measure": "",
             "see_also": "",
             "issue_tracker_item": "",
             "other": "",
             "comment": "",
         }
 
+        if hasattr(SSSOM_Slots, "similarity_score"):
+            expected_result["similarity_score"] = np.nan
+            expected_result["similarity_measure"] = ""
+        else:
+            expected_result["semantic_similarity_score"] = np.nan
+            expected_result["semantic_similarity_measure"] = ""
```

Comment:
Are you married to this? I can see why this would make sense in the framework code (to be backwards compatible with the 0.5.X lineage), but this is the test suite.. I'd be inclined to just fix the tests to use `similarity_score` instead?

Reply:
But if we do that, then the test will fail until (1) we publish the next version of the SSSOM schema (one that includes the `semantic_similarity_score` -> `similarity_score` renaming) and (2) SSSOM-Py’s dependencies are updated to depend on the new SSSOM schema.

And we can’t test the next version of the SSSOM schema until that changes is in, so we are in a kind of catch-22 here.

SSSOM-Py’s test suite currently can only pass with SSSOM-Schema <= 0.15.2. If we simply “fix the tests to use `similary_score` instead”, then it will only pass with the next version of the SSSOM-Schema, and will no longer pass with <= 0.15.2.

The way the PR does it, the test suite can accommodate both SSSOM-Schema <= 0.15.2 and the to-be-published version. It’s well worth the additional complexity of _one_ extra conditional instruction in my opinion.

Reply:
Sold. Thanks for explaining!